### ISSUE TITLE
`GradientBoostingClassifier` and `GradientBoostingRegressor` should support sparse matrices

### ISSUE THREAD

**Issue by SimengSun** [EducationalTestingService/skll#428]
MemoryError will be raised for large training data when using GBC (It converts sparse training data to dense first). GBC in sklearn does support sparse matrices as input and works well.

**Comment by desilinguist** [comment]
SKLL currently expects the following learners to take only dense matrices as input:

```
_REQUIRES_DENSE = (BayesianRidge,
                   GradientBoostingClassifier,
                   GradientBoostingRegressor,
                   Lars,
                   TheilSenRegressor)
```

I double checked and other than the gradient boosted learners, the others take dense matrices as expected.

**Comment by desilinguist** [comment]
Keep it open please.

**Comment by desilinguist** [comment]
Addressed by #429.
